src/transactions.ts

```diff
@@ -74,11 +74,22 @@ export class ReadTransactionImpl implements ReadTransaction {
  * WriteTransactions are used with `Replicache.register` and allows read and
  * write operations on the database.
  */
-export class WriteTransaction extends ReadTransactionImpl {
+export interface WriteTransaction extends ReadTransaction {
```

Comment:
Out of curiosity why have an interface? Are there two implementations now?

Reply:
Mostly from a documentation purpose. It also allows us to hide the constructor.

Reply:
*hide as in hide from the "public" API. If you really want to get it you can get it from the runtime representation of an object